tests/ImageSharp.Tests/Formats/Qoi/QoiDecoderTests.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace SixLabors.ImageSharp.Tests.Formats.Qoi;
+
+[Trait("Format", "Qoi")]
+[ValidateDisposedMemoryAllocations]
+public class QoiDecoderTests
+{
+    [Theory]
+    [InlineData(TestImages.Qoi.Dice)]
+    [InlineData(TestImages.Qoi.EdgeCase)]
+    [InlineData(TestImages.Qoi.Kodim10)]
+    [InlineData(TestImages.Qoi.Kodim23)]
+    [InlineData(TestImages.Qoi.QoiLogo)]
+    [InlineData(TestImages.Qoi.TestCard)]
+    [InlineData(TestImages.Qoi.TestCardRGBA)]
+    [InlineData(TestImages.Qoi.Wikipedia008)]
+    public void Identify(string imagePath)
+    {
+        TestFile testFile = TestFile.Create(imagePath);
+        using MemoryStream stream = new(testFile.Bytes, false);
+
+        ImageInfo imageInfo = Image.Identify(stream);
+
+        Assert.NotNull(imageInfo);
+        Assert.Equal(imageInfo.Metadata.DecodedImageFormat, ImageSharp.Formats.Qoi.QoiFormat.Instance);
+    }
+
+    [Theory]
+    [WithFile(TestImages.Qoi.Dice, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.EdgeCase, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.Kodim10, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.Kodim23, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.QoiLogo, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.TestCard, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.TestCardRGBA, PixelTypes.Rgba32)]
+    [WithFile(TestImages.Qoi.Wikipedia008, PixelTypes.Rgba32)]
+    public void Decode<TPixel>(TestImageProvider<TPixel> provider)
```

Comment:
We need at least a few test cases to validate `QoiMetadata` for the encoder & decoder.

Reply:
I'm not sure how could we determine the channels and color space when encoding image. They are purely informative and they don't affect how pixels are encoded, so there's no way to know if an image is RGA or RGBA

I though we could check if any pixel alpha is different to 255, then the channels will be RGBA, but some images (like `edgecase.qoi` or `testcard.qoi`) are fully opaque but their header say they're RGBA

Also I'm not sure how to create the `QoiMetadata` inside the decoder, so I though encoding every image with _RGBA_ channels (value 4 in enum) and _sRGB with linear alpha_ (value 0 in enum) since most images have that metadata

Reply:
Researching, I found https://github.com/phoboslab/qoi/issues/37, I guess it can give us more information since those bytes are purely informative but I didn't find how to decide what value it'll have

So I guess we could use the `PixelType` from the provider to decide it, what do you think?

Reply:
You can do a separate round-trip test: encode then re-decode with ImageSharp qoi codecs, and check the resulting `image.Metadata.GetQoiMetadata()`. This would test the encoder+decoder `QoiMetadata` handling functionality at the same time, so also address https://github.com/SixLabors/ImageSharp/pull/2446#discussion_r1251050456.

Reply:
Just noticed you did this already in [ef78f98](https://github.com/SixLabors/ImageSharp/pull/2446/commits/ef78f98eb3d686638f9da867f8896b286a722e10), so nevermind :)

Reply:
So that was correct?
I though we couldn't use any of the data from the tests :o